### PR TITLE
added hw 5.2 solution

### PR DESCRIPTION
**Pull request by programmer28**

### CODE REVIEW

File: src/main/java/core/basesyntax/Plane.java [hunk 1]
```diff
@@ -7,5 +7,54 @@
...
+            return this;
+        }
+
+        public PlaneBuilder setColor(String color) {
+            this.color = color;
+            return this;
+        }
+
+        public Plane build() {
+            return new Plane(model, speed, weight, age, color);
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Plane.java]:
```suggestion
            return new Plane(this);
```

File: src/main/java/core/basesyntax/Plane.java [hunk 2]
```diff
@@ -7,5 +7,54 @@
...
+    private String color;
 
+    public Plane(String model, int speed, int weight, int age, String color) {
+        this.model = model;
+        this.speed = speed;
+        this.weight = weight;
+        this.age = age;
+        this.color = color;
+    }
+
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Plane.java:23]: for correct Builder pattern, we need private constructor of Plane with a Builder object

File: src/main/java/core/basesyntax/Plane.java (hunk 1 above)
Comment (programmer28) [reply src/main/java/core/basesyntax/Plane.java]: Okay, I`ve fixed that

File: src/main/java/core/basesyntax/Plane.java (hunk 2 above)
Comment (programmer28) [reply src/main/java/core/basesyntax/Plane.java:23]: I`ve fixed that too.

**Review by Tarabarnik — APPROVED** [review]

**Pull request closed without merging**